src/richie/apps/search/indexers/courses.py

```diff
@@ -393,11 +395,13 @@ def get_es_document_for_course(cls, course, index=None, action="index"):
                 cmsplugin_ptr__position=0,
             ):
                 with translation.override(language):
-                    icon_images[language] = {
-                        **get_picture_info(icon, "icon"),
-                        "color": plugin_model.page.category.color,
-                        "title": plugin_model.page.get_title(),
-                    }
+                    picture_info = get_picture_info(icon, "icon")
+                    if picture_info:
+                        icon_images[language] = {
+                            **picture_info,
+                            "color": plugin_model.page.category.color,
+                            "title": plugin_model.page.get_title(),
```

Comment:
be careful you should still set the color and title even if the picture is not there!